### PR TITLE
Don't append a trailing comma to the last param in row

### LINKED ISSUE #96: Reformatting adds a trailing comma even after variadic params
If a method defintion gets reformatted, a comma is added after every parameter line - even after a variadic one. (hhvm 3.26.2)

```
use Facebook\HackCodegen\HackCodegenFactory;
use Facebook\HackCodegen\HackCodegenConfig;

$factory = new HackCodegenFactory(new HackCodegenConfig());

$class = $factory->codegenClass('Foo');
$class->addMethod(
  $factory->codegenMethod('bar')
    ->addParameter('int ...$baz_is_a_very_long_name_and_stuff_no_beer_and_no_fun_makes_me_a_dull_boy')
    ->setReturnType('void')
    ->setBody('echo "meh";')
);
```

Error:
```
Fatal error: Uncaught Error: A variadic parameter ('...') may not be followed by a comma. in test.hh:5
```

### PR DESCRIPTION
**Pull request by usox**
Fixes #96

### CODE REVIEW

File: src/CodegenFunctionish.php [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 namespace Facebook\HackCodegen;
 
-use namespace HH\Lib\{Str, Vec};
+use namespace HH\Lib\{Str, Vec, C};
```
Comment (fredemmott) [inline src/CodegenFunctionish.php]: alphabetical

**Comment by usox** [comment]
I was looking for a more generic solution because it feels a bit hacky to recognize variadics...but ok.

**Comment by fredemmott** [comment]
the reason is that one of the purposes of trailing commas is to preserve ‘git blame’ when adding new parameters/items

File: src/CodegenFunctionish.php [hunk 2]
```diff
@@ -164,7 +164,12 @@ protected function getFunctionDeclarationBase(
     ) {
       return (new HackBuilder($this->config))->add($code)->getCode();
     } else {
-      $parameter_lines = Vec\map($this->parameters, $line ==> $line.",");
+      $parameter_lines = Vec\map($this->parameters, $line ==> {
+        if (Str\search($line, '...$') !== null) {
```
Comment (fredemmott) [inline src/CodegenFunctionish.php]: https://3v4l.org/amPNQ

**Comment by usox** [comment]
Phew, i was not aware of that.... Maybe it's worth to make the params configurable instead of adding some special treatment for variadics in here...

**Comment by fredemmott** [comment]
We should consider refactoring parameters to not be string-based,

**Pull request merged**